src/recorder.cc

```diff
@@ -61,8 +69,13 @@ void Recorder::set_dump_count(unsigned int count) {
 Datum* Recorder::NewDatum(std::string title) {
   Datum* d = data_[index_];
   d->title_ = title;
-  d->vals_.resize(1);
-  d->shapes_.resize(1);
+  if (inject_sim_id_) {
```

Comment:
I think a ternary operator would be cleaner here, i.e. `int sz = inject_sim_id_ ? 1 : 0;`.